File: plugin/src/main/java/net/projecttl/economy/plugin/ProjectEconomy.java
package net.projecttl.economy.plugin;

import net.projecttl.economy.InitSQL;
import net.projecttl.economy.plugin.commands.EconomyCommand;
import net.projecttl.economy.plugin.listeners.RegisterListener;
import org.bukkit.ChatColor;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.Objects;
import java.util.logging.Logger;

public class ProjectEconomy extends JavaPlugin {

    private final Logger logger = this.getLogger();
    private final InitSQL database = new InitSQL(this);

    @Override
    public void onEnable() {
        if (!getDataFolder().exists()) {
            saveDefaultConfig();
        }

        logger.info(ChatColor.GREEN + "Project Economy plugin has enabled!");

        Objects.requireNonNull(getCommand("money")).setExecutor(new EconomyCommand(this));
        getServer().getPluginManager().registerEvents(new RegisterListener(this), this);
    }

    @Override
    public void onDisable() {
        logger.info(ChatColor.RED + "Project Economy plugin has disabled!");
        saveConfig();
    }

    public void connect() {
        database.connect();
    }

    public void disconnect() {
        database.disconnect();
    }
}

File: plugin/src/main/kotlin/net/projecttl/economy/plugin/ProjectEconomy.kt
package net.projecttl.economy.plugin

import net.projecttl.economy.plugin.listeners.RegisterListener
import net.projecttl.economy.plugin.utils.InitSQL
import org.bukkit.ChatColor
import org.bukkit.plugin.java.JavaPlugin

lateinit var instance: ProjectEconomy

class ProjectEconomy : JavaPlugin() {

    private val database = InitSQL(this)

    override fun onEnable() {
        instance = this

        if (!dataFolder.exists()) {
            this.saveDefaultConfig()
        }

        database.connect()

        getCommand("money")?.apply {
            setExecutor(CommandDispatcher)
            tabCompleter = CommandDispatcher
        }

        server.pluginManager.registerEvents(RegisterListener(), this)
        logger.info("${ChatColor.GREEN}Project Economy plugin has enabled!")
    }

    override fun onDisable() {
        logger.info("${ChatColor.RED}Project Economy plugin has disabled!")
        saveConfig()
        database.disconnect()
    }
}